src/mmw/js/src/modeling/tests.js

```diff
@@ -14,29 +14,39 @@ var _ = require('lodash'),
     App = require('../app.js'),
     modificationConfigUtils = require('./modificationConfigUtils');
 
+var sandboxId = 'sandbox',
+    sandboxSelector = '#' + sandboxId;
+
 describe('Modeling', function() {
     before(function() {
-        if ($('#sandbox').length === 0) {
-            $('<div>', {id: 'sandbox'}).appendTo('body');
+        if ($(sandboxSelector).length === 0) {
+            $('<div>', {id: sandboxId}).appendTo('body');
         }
+    });
 
+    beforeEach(function() {
         // ScenarioModel.initialize() expects
         // App.currProject to be set, and uses it to determine the
         // taskModel and modelPackage to use.
         App.currProject = new models.ProjectModel();
-    });
 
-    beforeEach(function() {
         this.server = sinon.fakeServer.create();
         this.server.respondImmediately = true;
+
+        // Use fake debounce so that things will execute more synchronously.
+        this.origDebounce = _.debounce;
+        _.debounce = _.identity;
```

Comment:
I'll put this in a comment:

Using debounce causes adding modifications to become asynchronous due to using setTimeout, which makes it more difficult to test. Using the fake timer in sinon does not help because lodash has its own cached copy of the setTimeout function.